FrontBundle/Controller/NodeController.php

```diff
@@ -106,11 +107,32 @@ public function previewAction($token)
         $siteManager = $this->get('open_orchestra_display.manager.site');
         $siteManager->setSiteId($node->getSiteId());
         $siteManager->setCurrentLanguage($node->getLanguage());
+        $this->updatePreviewRequestParameters($request, $node);
 
         return $this->renderNode($node, array('token' => $token));
     }
 
     /**
+     * @param Request           $request
+     * @param ReadNodeInterface $node
+     */
+    protected function updatePreviewRequestParameters(Request $request, ReadNodeInterface $node)
+    {
+        $routeParams = $request->get('_route_params');
+        $routeParams = array_merge(array(
+            'siteId' => $node->getSiteId(),
+            'nodeId' => $node->getNodeId(),
+            '_locale' => $node->getLanguage()
+        ),
+            $routeParams
```

Comment:
@alavieille : you should check that $routeParams is a valid array before merging to avoid a sad 500 error